core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -249,6 +249,9 @@ CompletionStage<Configuration> createCache(String cacheName, String template, Co
       } catch (Exception e) {
          throw CONFIG.configurationSerializationFailed(cacheName, configuration, e);
       }
+      if (cacheName.length() > 255) {
```

Comment:
I would move this to be the first check in the method so that we don't have to serialize the configuration unnecessarily.